pkg/server/api/bundle/v1/service_test.go

```diff
@@ -960,85 +977,53 @@ func TestListFederatedBundles(t *testing.T) {
 			},
 		},
 		{
-			name:          "get only trust domains",
-			expectBundles: []*common.Bundle{b1, b2, b3},
-			outputMask:    &types.BundleMask{},
-		},
-		{
-			name: "get first page",
+			name: "page bundles",
 			// Returns only one element because server bundle is the first element
 			// returned by datastore, and we filter resutls on service
-			expectBundles: []*common.Bundle{b1},
-			expectToken:   serverTrustDomain.String(),
-			pageSize:      2,
-		},
-		{
-			name:          "get second page",
-			expectBundles: []*common.Bundle{b2, b3},
-			expectToken:   td3.String(),
-			pageSize:      2,
-			pageToken:     serverTrustDomain.String(),
-		},
-		{
-			name:          "get third page",
-			expectBundles: []*common.Bundle{},
-			expectToken:   "",
-			pageSize:      2,
-			pageToken:     td3.String(),
-		},
-		{
-			name: "datastore returns invalid trust domain",
-			code: codes.Internal,
-			err:  `bundle has an invalid trust domain ID: "invalid TD"`,
-			expectLogs: []spiretest.LogEntry{
-				{
-					Level:   logrus.ErrorLevel,
-					Message: "Bundle has an invalid trust domain ID",
-					Data: logrus.Fields{
-						logrus.ErrorKey:         `spiffeid: unable to parse: parse spiffe://invalid TD: invalid character " " in host name`,
-						telemetry.TrustDomainID: "invalid TD",
-					},
-				},
+			expectBundlePages: [][]*common.Bundle{
+				{b1},
+				{b2, b3},
+				{},
 			},
-			isInvalidTD: true,
+			pageSize: 2,
 		},
 	} {
 		tt := tt
 		t.Run(tt.name, func(t *testing.T) {
 			test.logHook.Reset()
 
-			// Create an invalid bundle to test mask failing
-			if tt.isInvalidTD {
-				invalidBundle := createBundle(t, test, "invalid TD")
-				defer func() {
-					_, _ = test.ds.DeleteBundle(ctx, &datastore.DeleteBundleRequest{
-						TrustDomainId: invalidBundle.TrustDomainId,
-					})
-				}()
-			}
-
-			resp, err := test.client.ListFederatedBundles(ctx, &bundlepb.ListFederatedBundlesRequest{
-				OutputMask: tt.outputMask,
-				PageSize:   tt.pageSize,
-				PageToken:  tt.pageToken,
-			})
-
-			if tt.err != "" {
-				spiretest.RequireGRPCStatusContains(t, err, tt.code, tt.err)
-				require.Nil(t, resp)
-				spiretest.AssertLogs(t, test.logHook.AllEntries(), tt.expectLogs)
+			var pageToken string
+			var actualBundlePages [][]*types.Bundle
+			for {
```

Comment:
may we add a limit here? just to prevnet we never got pageToken == "" by error

Reply:
Yep. Done.